code/game/objects/items/weapons/melee/melee_misc.dm

```diff
@@ -45,6 +45,391 @@
+	righthand_file = 'icons/mob/inhands/weapons_righthand.dmi'
+	icon_state = "secsword0"
+	item_state = "secsword0"
+	flags = CONDUCT
+	force = 15
+	throwforce = 5
+	w_class = WEIGHT_CLASS_BULKY
+	sharp = TRUE
+	origin_tech = "combat=4"
+	attack_verb = list("stabbed", "slashed", "sliced")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	materials = list(MAT_METAL = 1000)
+	needs_permit = TRUE
+	///The icon the sword has when turned off
+	var/base_icon = "secsword"
+	///How much stamina damage the sword does in stamina mode
+	var/stam_damage = 35 // 3 hits to stamcrit
+	///How much burn damage the sword does in burn mode
+	var/burn_damage = 10
+	/// How much power does it cost to stun someone
+	var/stam_hitcost = 1000
+	/// How much power does it cost to burn someone
+	var/burn_hitcost = 1500
+	/// the initial cooldown tracks the time between stamina damage. tracks the world.time when the baton is usable again.
+	var/cooldown = 3.5 SECONDS
+	///The sword's power cell - starts high
+	var/obj/item/stock_parts/cell/high/cell = null
+	///The sword's current mode. Defaults to off.
+	var/state = SECSWORD_OFF
+
+/obj/item/melee/secsword/examine(mob/user)
+	. = ..()
+	if(!cell)
+		. += "<span class='notice'>The powercell has been removed!</span>"
+		return
+	. += "<span class='notice'>It is [round(cell.percent())]% charged.</span>"
+	if(round(cell.percent() < 100))
+		. += "<span class='notice'>Can be recharged with a recharger.</span>"
+
+/obj/item/melee/secsword/examine_more(mob/user)
+	. = ..()
+	. += "A simple, practical blade developed by Shellguard munitions for ‘enhanced’ riot control."
+	. += ""
+	. += "The Securiblade is a simple blade of lightweight silver plasteel, honed to a nano-thin edge and with a simple \
+	power pack built into the hilt. A multi-purpose option for the NT Officer with a flair for impracticality, the \
+	Securiblade serves just as well as a regular sword, as it does a stun weapon or even an energized, heated blade."
+	. += ""
+	. += "While not the most popular option among the officers of Nanotrasen’s security forces, the Securiblade has still been valued for the multiple options it \
+	presents in combat. Deactivated, it’s a simple sword, but powered, it can either be utilized as a useful stun weapon, or as a dangerous, heated blade \
+	that can inflict grievous burn wounds on any suspects unfortunate enough to meet an officer using it. Rest assured, the Securiblade is a reliable tool in the hands of a skilled officer."
+
+/obj/item/melee/secsword/Destroy()
+	if(cell?.loc == src)
+		QDEL_NULL(cell)
+	return ..()
+
+/obj/item/melee/secsword/update_icon_state()
+	if(!cell)
+		icon_state = "[base_icon]3"
+		item_state = "[base_icon]0"
+	else
+		icon_state = "[base_icon][state]"
+		item_state = "[base_icon][state]"
+
+/obj/item/melee/secsword/Initialize(mapload)
+	. = ..()
+	link_new_cell()
+	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 1, _parryable_attack_types = NON_PROJECTILE_ATTACKS)
```

Comment:
Can you move this to be right under the type define? And the `Destroy()` proc to be right under this one?